cirq/ops/common_channels.py

```diff
@@ -187,3 +189,603 @@ def depolarize(p: float) -> DepolarizingChannel:
+    """
+    return AmplitudeDampingChannel(gamma)
+
+
+class PhaseDampingChannel(raw_types.Gate):
+    """
+    This channel models phase damping which is the loss of quantum
+    information without the loss of energy.
+    """
+
+    def __init__(self, gamma) -> None:
+        """
+        Construct a channel that enacts phase damping at the rate
+        given by gamma
+
+        This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+        With
+            M_0 = \begin{bmatrix}
+                    1 & 0 \\
+                    0 & \sqrt{1 - \gamma}
+                  \end{bmatrix}
+            M_1 = \begin{bmatrix}
+                    0 & 0 \\
+                    0 & \sqrt{\gamma}
+                  \end{bmatrix}
+
+        Args:
+            gamma: The damping rate
+
+        Raises:
+            ValueError is gamma is not a valid probability.
+        """
+
+        def validate_probability(p, p_str):
+            if p < 0:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._gamma = validate_probability(gamma, 'gamma')
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        return (
+            np.array([[1., 0.], [0., np.sqrt(1. - self._gamma)]]),
+            np.array([[0., 0.], [0., np.sqrt(self._gamma)]]),
+        )
+
+    def _eq_tuple(self):
+        return (PhaseDampingChannel, self._gamma)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.phase_damp(gamma={!r})'.format(self._gamma)
+
+    def __str__(self) -> str:
+        return 'phase_damp(gamma={!r})'.format(self._gamma)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PD({!r})'.format(self._gamma)
+
+
+def phase_damp(gamma: float) -> PhaseDampingChannel:
+    """
+    Creates a PhaseDampingChannel with damping rate gamma
+
+    This channel evolves a density matrix via
+           \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \begin{bmatrix}
+                1 & 0  \\
+                0 & \sqrt{1 - \gamma}
+              \end{bmatrix}
+        M_1 = \begin{bmatrix}
+                0 & 0 \\
+                0 & \sqrt{\gamma}
+              \end{bmatrix}
+
+    Args:
+        gamma: The damping rate
+
+    Raises:
+        ValueError is gamma is not a valid probability.
+    """
+    return PhaseDampingChannel(gamma)
+
+
+class PhaseFlipChannel(raw_types.Gate):
+    """
+    Flips the sign of the phase of a qubit with a given probability.
+    """
+
+    def __init__(self, p) -> None:
+
+        """
```

Comment:
Style guide is to put a short one-line summary on the same line as the `"""` before moving on to the more complicated body.

Reply:
fixed.